### PR TITLE
HDDS-815. Rename Ozone/HDDS config keys prefixed with 'dfs'

### PR DESCRIPTION
**Pull request by sarvekshayr**
## What changes were proposed in this pull request?
Renaming config keys and its corresponding variable name prefixed with  'dfs'  to 'ozone'.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-815

## How was this patch tested?
Patch was tested by running the test classes of the modified files.

### CODE REVIEW

**Review by tanvipenumudy — COMMENTED** [review]
Thank you @sarvekshayr for the patch. 
Could you please rebase your patch on top of the latest master?

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineStream.java [hunk 1]
```diff
@@ -113,20 +112,18 @@ public void setup() throws Exception {
...
-        .setStreamBufferMaxSize(MAX_FLUSH_SIZE)
-        .applyTo(conf);
-
     conf.setLong(OzoneConfigKeys.DFS_RATIS_SNAPSHOT_THRESHOLD_KEY, 1);
     conf.setQuietMode(false);
     cluster =
         MiniOzoneCluster.newBuilder(conf)
             .setNumDatanodes(3)
             .setHbInterval(200)
+            .setDataStreamMinPacketSize(1024)
```
Comment (tanvipenumudy) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineStream.java:121]: The changes made under this test class seem unintended, could you please address these?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java [hunk 2]
```diff
@@ -36,8 +36,8 @@
 @InterfaceAudience.Public
 @InterfaceStability.Unstable
 public final class OzoneConfigKeys {
-  public static final String DFS_CONTAINER_IPC_PORT =
-      "dfs.container.ipc";
+  public static final String OZONE_CONTAINER_IPC_PORT =
+      "ozone.container.ipc.port";
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:40]: To reduce the size of the patch, I suggest renaming items in two steps:

1. config keys (`dfs.container.ipc` to `ozone.container.ipc.port`)
2. java constants (`DFS_CONTAINER_IPC_PORT` to `OZONE_CONTAINER_IPC_PORT`)

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sarvekshayr for working on this.  When changing config keys, we have to consider backwards compatibility:

Existing clusters may already be configured using the old keys.  If we simply change the keys, those configs will no longer work after upgrade.  Cluster will suddenly start to use the default settings.

The way to keep old settings work is by deprecating the old key, which is usually done here:

https://github.com/apache/ozone/blob/60bcdafa73affe68d3ebed5a74e7b54a414efcf4/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java#L313-L328

Also, please wait for clean [CI run](https://github.com/sarvekshayr/ozone/actions/runs/7812765179) in your fork before opening PRs.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineStream.java [hunk 3]
```diff
@@ -114,14 +114,14 @@ public void setup() throws Exception {
...
-        .setStreamBufferMaxSize(MAX_FLUSH_SIZE)
-        .applyTo(conf);
-
-    conf.setLong(OzoneConfigKeys.DFS_RATIS_SNAPSHOT_THRESHOLD_KEY, 1);
+            .setDataStreamMinPacketSize(1024)
+            .setBlockSize(BLOCK_SIZE)
+            .setChunkSize(CHUNK_SIZE)
+            .setStreamBufferFlushSize(FLUSH_SIZE)
+            .setStreamBufferMaxSize(MAX_FLUSH_SIZE)
+            .applyTo(conf);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineStream.java:122]: nit: please keep original indent

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java (hunk 2 above)
Comment (sarvekshayr) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:40]: Added backward compatibility, made all the necessary fixes and got a clean CI run.
I'll keep in my mind about reducing the size of the patch in the coming PRs.

**Comment by sarvekshayr** [comment]
> Thanks @sarvekshayr for updating the patch.
> 
> There are a few config keys with `dfs.` prefix left. Probably the list in the Jira issue is outdated.
> 
> ```
> hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmConfigKeys.java
> 54:      = "dfs.container.ratis.num.write.chunk.threads.per.volume";
> 82:      "dfs.container.ratis.statemachinedata.sync.retries";
> 85:      "dfs.container.ratis.statemachine.max.pending.apply-transactions";
> 92:      "dfs.container.ratis.log.queue.num-elements";
> 96:      "dfs.container.ratis.log.queue.byte-limit";
> 101:      "dfs.container.ratis.log.appender.queue.num-elements";
> 105:      "dfs.container.ratis.log.appender.queue.byte-limit";
> 109:      "dfs.container.ratis.log.purge.gap";
> 114:      "dfs.container.ratis.leader.pending.bytes.limit";
> 119:      "dfs.ratis.server.retry-cache.timeout.duration";
> 
> hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java
> 68:      "dfs.container.ratis.datastream.random.port";
> 74:      "dfs.container.chunk.write.sync";
> 86:      "dfs.container.ratis.admin.port";
> 92:      "dfs.container.ratis.server.port";
> 99:      = "dfs.container.ratis.datastream.enabled";
> 103:      = "dfs.container.ratis.datastream.port";
> ```
> 
> Also, Java constants for default values of the renamed config keys still use the old `DFS_` prefix. It would be nice to match names for key and default value. Example:
> 
> https://github.com/apache/ozone/blob/2cbabcc090dc6f78b0af8c4221c777b550bdc92b/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java#L79-L81

Sure! I'll rename the remaining ones as well. I'll create a sub-task to do the same in order to reduce the size of this patch.

**Comment by adoroszlai** [comment]
To clarify my comment about separating changes: it's not just about the size of the patch itself.

Renaming config keys is a relatively small change, focused on a few files, but it must be well tested to ensure compatibility.

On the other hand, renaming constants is a massive, but simple change, affecting ~50 files in this case.  Yet, regular CI checks are enough as a test; it should be fine as long as code compiles and checkstyle is happy.  This part of the change is likely to cause merge conflicts for other PRs.

**Comment by sarvekshayr** [comment]
Thank you for the clarification. I'll make the changes as indicated.

**Comment by adoroszlai** [comment]
Updated HDDS-815 to reflect the current set of config keys to be renamed (6 keys no longer exist, 15 new keys).

The original task description mentioned using "either HDDS or Ozone" as prefix.  I have updated it to use `hdds.` based on:
- these keys belong to Datanode, which is a HDDS service (`hadoop-hdds/container-service`)
- more existing keys are named `hdds.container...` than `ozone.container...`

**Pull request closed without merging**